sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -252,12 +256,23 @@ public Path getPath(URI uri) {
     }
 
     /**
-     * Unsupported. Use {@link #newInputStream(Path, OpenOption...)} or {@link #newOutputStream(Path, OpenOption...)}
-     * instead.
+     * Opens or creates a file, returning a seekable byte channel to access the file.
+     * <p>
+     * This method is primarily offered to support some jdk convenience methods such as
+     * {@link Files#createFile(Path, FileAttribute[])} which requires opening a channel and closing it. A channel may
+     * only be opened in read mode OR write mode. It may not be opened in read/write mode. Seeking is supported for
+     * reads, but not for writes. Modifications to existing files is not permitted--only creating new files or
+     * overwriting existing files.
+     * <p>
+     * This type is not threadsafe to prevent having to hold locks across network calls.
+     * <p>
+     * Use {@link #newInputStream(Path, OpenOption...)} or {@link #newOutputStream(Path, OpenOption...)}
+     * instead. Please see the documentation on these methods for information on how to use this type. This type can
+     * only be opened in read OR write mode. Read/write mode is not supported.
      *
-     * @param path the Path
-     * @param set open options
-     * @param fileAttributes attributes
+     * @param path the path of the file to open
+     * @param set options specifying how the file should be opened
+     * @param fileAttributes an optional list of file attributes to set atomically when creating the directory
      * @return a new seekable byte channel
      * @throws UnsupportedOperationException Operation is not supported.
```

Comment:
This isn't thrown anymore from what I can tell

Reply:
It can still be thrown by either newOutputStream or newInputStream if they specify an option we don't support.